### PR TITLE
Implementing the TokenAcquirerFactory methods

### PR DESCRIPTION
**Pull request by jmprieur**

### CODE REVIEW

File: TokenAcquirerTests/TokenAcquirer.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public async Task AcquireTokenWithPop_ClientCredentialsAsync()
             var cert = options.ClientCredentials.First().Certificate;
 
             // Get the token acquisition service
-            ITokenAcquirer tokenAcquirer = serviceProvider.GetRequiredService<ITokenAcquirer>();
+            ITokenAcquirer tokenAcquirer = tokenAcquirerFactory.GetTokenAcquirer(string.Empty);
```
Comment (jennyf19) [inline TokenAcquirerTests/TokenAcquirer.cs:106]: do we need to support the case with no authentication scheme?

File: src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs [hunk 2]
```diff
@@ -371,6 +374,20 @@ internal static void UpdateConfidentialClientApplicationOptionsFromMergedOptions
...
+        {
+            if (string.IsNullOrEmpty(mergedOptions.TenantId) && string.IsNullOrEmpty(mergedOptions.Instance) && !string.IsNullOrEmpty(mergedOptions.Authority))
+            {
+                string authority = mergedOptions.Authority.TrimEnd('/');
+                int indexTenant = authority.LastIndexOf('/');
+                if (indexTenant >= 0)
+                {
+                    mergedOptions.Instance = authority.Substring(0, indexTenant);
+                    mergedOptions.TenantId = authority.Substring(indexTenant + 1);
+                }
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs:387]: maybe we can do better here, but should not hold the PR

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirer.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
+﻿using System;
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirer.cs:1]:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

﻿using System;
```

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirer.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+using System.Threading;
+
+namespace Microsoft.Identity.Web
+{
+    internal class TokenAcquirer : ITokenAcquirer
+    {
+        private readonly ITokenAcquisition _tokenAcquisition;
+        private readonly string _authenticationScheme;
+
+        public TokenAcquirer(ITokenAcquisition tokenAcquisition, string authenticationScheme)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirer.cs:15]: this is cool

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs [hunk 5]
```diff
@@ -1,12 +1,12 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
+﻿// Licensed under the MIT License.
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs:1]:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
```

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs [hunk 6]
```diff
@@ -157,7 +179,56 @@ public ITokenAcquirer GetTokenAcquirer(string authority, string region, string c
...
+                var tokenAcquisition = ServiceProvider.GetRequiredService<ITokenAcquisition>();
+                acquirer = new TokenAcquirer(tokenAcquisition, authenticationScheme);
+                authSchemes.Add(authenticationScheme, acquirer);
+            }
+            return acquirer;
+        }
+
+        private static string GetKey(string? authority, string? clientId)
+        {
+            return $"{authority}{clientId}";
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs:231]: not including the region any longer?

File: tests/daemon-app/Daemon-app/Program - SDK.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                region: null,
+                certificate: CertificateDescription.FromKeyVault("https://webappsapistests.vault.azure.net", "Self-Signed-5-5-22")
+                );
+
+            // Get the token acquisition service
+            var result = await tokenAcquirer.GetTokenForAppAsync("https://graph.microsoft.com/.default");
+            Console.WriteLine($"Token expires on {result.ExpiresOn}");
+
+            result = await tokenAcquirer.GetTokenForAppAsync("https://graph.microsoft.com/.default");
+            Console.WriteLine($"Token expires on {result.ExpiresOn}");
```
Comment (jennyf19) [inline tests/daemon-app/Daemon-app/Program - SDK.cs:56]: this is great.

**Review by jennyf19 — APPROVED** [review]
:shipit: 
few minor comments which can be covered in another PR. Great work!

File: TokenAcquirerTests/TokenAcquirer.cs (hunk 1 above)
Comment (jmprieur) [reply TokenAcquirerTests/TokenAcquirer.cs:106]: I renamed the parameter to optionName, and set it to string.Empty by default.

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs (hunk 6 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs:231]: I dont' think that a same application deployed instance could run in different regions.

**Pull request merged**